transcom/mymove
# B 18111 pay grade and duty location flow changes

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3a871310)

## Summary

This story moves the entry of grade and current duty location to the orders page. Please see the documents attached to the story and associated feature as they are really helpful.

With this story, the service member rank is still being maintained and updated to be the order grade value, which will make sure nothing is broken that is still using the service member rank.

As of right now, both the Profile and the Orders page will show the pay grade and current duty location, and can be edited from both spots. Editing from the profile may change in the future, depending on what the gov would like to see.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### How to test

1. Access the milmove app
2. Login as a customer
3. Create a new profile, running through the flow. Do not add orders yet.
4. After creating the profile and before adding orders, go to the Edit Profile page and see that there is no information for pay grade and current duty location. 
5. Verify that you are not asked for pay grade and current duty location until adding your orders.
6. Add orders, entering pay grade and current duty location.
7. After entering, go to the database and verify that the order grade and order origin duty location match the information you provided.
8. Important: Also go to the service member table and verify that the rank and current location of the service member match the orders values you provided.
9. Edit the orders pay grade and orders current duty location.
10. Verify that in the database, both the service member and orders values are updated.
11. (Optional) As of right now, the customer can also edit the orders values from the Profile page.
